llama-index-core/llama_index/core/agent/runner/base.py

```diff
@@ -395,7 +395,7 @@ def _run_step(
         """Execute step."""
         dispatch_event = dispatcher.get_dispatch_event()
 
-        dispatch_event(AgentRunStepStartEvent())
+        dispatch_event(AgentRunStepStartEvent(task_id=task_id, step=step, input=input))
```

Comment:
hehe so adding custom kwargs/tags to events were pretty easy, right? I forgot the interface for `dispatch.event()` took into `BaseEvent` meaning we can define different construction params for the various events.

Reply:
I was thinking custom tags/kwags would be done at the per-handler-instance level 🤔 Hence the motivation for that context-manager-like design

Reply:
ahhhh right....my confusion!

Reply:
unless we allow the user to pass in tags in various modes? Something like:

- `str` -> blanket
- `dict` -> can specify the tags for specific events

Reply:
Hmm true true. I will figure this out in a follow-up PR 💪🏻